src/basic/chase.c

```diff
@@ -640,13 +680,10 @@ int chase_and_open(const char *path, const char *root, ChaseFlags chase_flags, i
                 return r;
         assert(path_fd >= 0);
 
-        assert_se(q = path_startswith(p, empty_to_root(root)));
-        if (isempty(q))
-                q = ".";
-
-        if (!FLAGS_SET(chase_flags, CHASE_PARENT)) {
-                r = path_extract_filename(q, &fname);
-                if (r < 0 && r != -EADDRNOTAVAIL)
+        if (!FLAGS_SET(chase_flags, CHASE_PARENT) &&
+            !FLAGS_SET(chase_flags, CHASE_EXTRACT_FILENAME)) {
```

Comment:
`!FLAGS_SET(chase_flags, CHASE_PARENT|CHASE_EXTRACT_FILENAME)`?

Reply:
No, filename needs to be extracted only when both flags are not set.
Note, `FLAGS_SET(flags, FLAG_A | FLAG_B)` and `FLAGS_SET(flags, FLAG_A) || FLAGS_SET(flags, FLAG_B)` are different.